src/test/groovy/transform/stc/GenericsSTCTest.groovy

```diff
@@ -1800,6 +1800,19 @@ assert result == 'ok'
 '''
     }
 
+    // GROOVY-7742
+    void testMethodParamWithDifferentPlaceholderForTypeArgument() {
+        // We use <T> here while List uses <E>
```

Comment:
Looks like when it resolves the placeholders it adds T based on DGM `static <T> T first(List<T> self)` since it uses `T` as placeholder.  The real code that produced this error used `collect` which also uses `T`.  Changing the code to use `return classes.get(0)` fixes it.